README.md

```diff
@@ -1,5 +1,37 @@
 # Rancher Catalog
 
+In Rancher, one of the two automatically packaged catalogs is this repository and the [community-catalog](https://github.com/rancher/community-catalog). The **library** catalog points to a git mirror of the **master** branch of this repo.
+
+## Branches
+
+When developing and testing new templates, a new dev branch for the upcoming release is created and packaged in our `rancher/server:master` container. For example, when we are developing and testing for `v1.6.0`, a `v1.6.0-dev` branch is created. Any PRs with changes should be made to this **dev** branch.
+
+Each Rancher RC is packaged with a specific branch to isolate what specific versions of templates were tested. If there are fixes that need to be made to address the templates, they should always be made to the **dev** branch. QA will pick up the **dev** branch for their RCs, if there are known issues in the specific branch packaged for the RC.
+
+> **Note:** Branches for specific RCs may be deleted in the future, as Rancher does not maintain the RCs.
+
+Before a Rancher release is created, a PR will be made to bring all changes from the **dev** branch to the **master** branch. The commits will be squashed into the 1 commit.
+
+## Versioning templates
+
+When making changes to a template into an existing catalog, compare the current **dev** branch with the **master** branch to see if a new folder needs to be created.
+
+1. Check to see if a new folder needs to be created or not.
+
+  a. If there are no new folders, a new folder will need to be created. Create a base copy of the latest default.
+      * Determine what the current default version of the catalog is. **IT IS NOT GUARANTEED TO BE THE LATEST FOLDER.** Go to the `config.yml` of the template to find the version number. Find the corresponding template folder that has this version.
+      * Copy the folder and commit the change as `Base copy of folder X`
```

Comment:
An example story:

Let's take a scenario from the 1.5 release for say ipsec folder:

When the 1.5-dev branch was created, the folder was 4. The image was rancher/net:v0.6.3. So now in the dev branch, after a bug fix we have a new `ranchet/net:v0.7.0`. Does this go in the same folder or a new one? Since this is the first time, we need a new folder here. So we do `cp -r ipsec/4 ipsec/5` and commit it as base copy. Now edit the image in ipsec to `ranchet/net:v0.7.0`. Now before 1.5 release is made, if you have newer releases like v0.7.1, v0.7.2, should we make a new folder or use the same folder 5?